nalgebra-sparse/src/cs.rs

```diff
@@ -569,12 +569,10 @@ where
             value_refs = values;
         }
         None => {
-            if sort && minor_indices.len() > 0 {
-                return Err(SparseFormatError::from_kind_and_msg(
-                    SparseFormatErrorKind::InvalidStructure,
-                    "No values provided for sorting.",
-                ));
-            }
+            assert!(
+                sort && minor_indices.len() > 0,
+                "No values provided for sorting."
+            );
```

Comment:
Probably I misunderstood your last comment on these code lines when you said that it would be OK to have this restriction so far (see https://github.com/dimforge/nalgebra/pull/1015#discussion_r766490039). So I just replaced error response with a panic. 
So should we ignore a case when values are absent?

Reply:
Right. Yes, panicking in this case is fine, but I'm still confused about the assert condition, as it seems to do the opposite of the previous error, doesn't it? Also, I don't understand why we test for `minor_indices.len() > 0`?

Isn't something like this what you really want?

```rust
None => {
   if sort {
      panic!("Internal error: Currently don't support sorting if no values are present.");
   }
}
```
Perhaps I'm missing something though!

Reply:
If `minor_indices.len() = 0` there would be nothing to sort, I suppose. So there would be no need to panic.
And I think in the previous error condition the assertion was the same (see [link](https://github.com/dimforge/nalgebra/pull/1015/commits/9ff5d1a97929d4a9494383eb6dc5590f5243491f#diff-ad9fa9a189df6658e5fee89093cd2efba6292df75401887d4c3e336a6fca2e4bL572) ).
But I see what you mean now, I used assert!() in a wrong way. Thank you.